### PR TITLE
chore: run tests on macOS

### PR DESCRIPTION
**Pull request by mkenigs**
- enable tests on macOS and make the naming of jobs more clear
- add x86_64-darwin case to upgrade test
- disable develop tests
- disable some publish tests

Getting this green has taken long enough that I think it's probably better to split out fixing all the tests into https://github.com/flox/flox/issues/277 and https://github.com/flox/flox/issues/278

Pinging @flox/cli if anyone wants to review

### CODE REVIEW

**Comment by mkenigs** [comment]
@bryanhonof just to double check, we don't want to specify runners, right?

**Comment by bryanhonof** [comment]
> @bryanhonof just to double check, we don't want to specify runners, right?

Correct, not yet.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,15 +10,16 @@ on:
...
       matrix:
-        include:
-          - package: flox
-          - package: flox-bash
-          - package: nix-editor
+        package:
+        - {name: flox, job_description: "Build and test"}
+        - {name: flox-bash, job_description: "Build" }
+        - {name: nix-editor, job_description: "Build" }
+        os: [ubuntu-latest, macos-latest]
```
Comment (bryanhonof) [inline .github/workflows/ci.yml:22]:
```suggestion
        os: 
        - ubuntu-latest
        - macos-latest
```
Just nitpicking, but it'd be more consistent with the others.

**Review by bryanhonof — APPROVED** [review]

**Comment by mkenigs** [comment]
No user facing changes

**Pull request merged**